src/record/record-transition.js

```diff
@@ -147,6 +148,11 @@ RecordTransition.prototype.add = function (socketWrapper, version, message) {
       return
     }
 
+    if(!utils.isOfType(data, 'object') && !utils.isOfType(data, 'array')) {
```

Comment:
There are several libraries that provide more complete type checking, how far do you want to do take it?